icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/CjMLBreakEngine.java

```diff
@@ -0,0 +1,346 @@
+            featureList.add(sb.append("TB3:").append(b3).append(b4).append(b5).toString());
+        }
+        if (isValid(b4) && isValid(b5) && isValid(b6)) {
+            sb.setLength(0);
+            featureList.add(sb.append("TB4:").append(b4).append(b5).append(b6).toString());
+        }
+        int score = fNegativeSum;
+        for (int j = 0; j < featureList.size(); j++) {
+            String key = featureList.get(j);
+            if (fModel.containsKey(key)) {
+                score += (2 * fModel.get(key));
+            }
+        }
+        if (score > 0) {
+            boundary.add(index);
+        }
+    }
+
+    private static boolean isValid(String str) {
+        return str.length() != 1 || str.codePointAt(0) != INVALID;
+    }
+
+    /**
+     * Initialize the element list from the input string.
+     *
+     * @param inString A input string to be segmented.
+     * @param elementList A list to store the first six characters and their unicode block codes.
+     * @param numCodePts The number of code points of input string
+     */
+    private void initElementList(CharacterIterator inString, ArrayList<Element> elementList,
+                                 int numCodePts) {
+        inString.setIndex(0);
+        final int w1 = INVALID;
+        final int w2 = INVALID;
+        final int w3 = current32(inString);
+        final int w4 = next32(inString);
+        final int w5 = 2 < numCodePts ? next32(inString) : INVALID;
+        final int w6 = 3 < numCodePts ? next32(inString) : INVALID;
+
+        final String b1 = String.valueOf((char) INVALID);
+        final String b2 = b1;
+        final String b3 = getUnicodeBlock(w3);
+        final String b4 = getUnicodeBlock(w4);
+        final String b5 = getUnicodeBlock(w5);
+        final String b6 = getUnicodeBlock(w6);
+
+        elementList.add(new Element(w1, b1));
+        elementList.add(new Element(w2, b2));
+        elementList.add(new Element(w3, b3));
+        elementList.add(new Element(w4, b4));
+        elementList.add(new Element(w5, b5));
+        elementList.add(new Element(w6, b6));
+    }
+
+    /**
+     * Get the character's unicode block code defined in UBlockCode.
+     *
+     * @param ch A char.
+     * @return The unicode block code which is 3 digits with '0' added in the beginning if the code
+     * is less than 3 digits.
+     */
+    private String getUnicodeBlock(int ch) {
+        UCharacter.UnicodeBlock block = UCharacter.UnicodeBlock.of(ch);
+        if (block == UCharacter.UnicodeBlock.NO_BLOCK
+                || block == UCharacter.UnicodeBlock.INVALID_CODE) {
+            return String.valueOf((char) INVALID);
+        } else {
+            return String.format("%03d", block.getID());
+        }
+    }
+
+    /**
+     * Load the machine learning's model file.
+     */
+    private void loadMLModel() {
+        int index = 0;
+        UResourceBundle rb = UResourceBundle.getBundleInstance(ICUData.ICU_BRKITR_BASE_NAME, "jaml");
```

Comment:
Question: I think this throws an exception if the data isn't present; is that the intended behavior?

Reply:
It already throws MissingResourceException If no resource bundle for the specified base name can be found.
https://github.com/unicode-org/icu/blob/main/icu4j/main/classes/core/src/com/ibm/icu/util/UResourceBundle.java#L364